<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form_SpecViewer1</class>
 <widget class="QWidget" name="Form_SpecViewer1">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>867</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QLabel" name="instruction_label">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="acceptDrops">
      <bool>true</bool>
     </property>
     <property name="text">
      <string>Drag and Drop IQ files Here</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="FormB" native="true">
     <layout class="QVBoxLayout" name="FormCLayout">
      <item>
       <widget class="QGroupBox" name="DataFormatGroup">
        <property name="title">
         <string>Data Format</string>
        </property>
        <layout class="QFormLayout" name="DataFormatGroupLayout">
         <property name="verticalSpacing">
          <number>3</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>3</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="header_label">
           <property name="text">
            <string>Header Offset (Bytes)</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="header_input">
           <property name="maximum">
            <number>2147483647</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="bandwidth_label">
           <property name="text">
            <string>Bandwidth (Hz):</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="bandwidth_input">
           <property name="maximum">
            <number>2147483647</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="isReal_label">
           <property name="text">
            <string>isReal?</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="isReal_input">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="samplingrate_label">
           <property name="text">
            <string>Sampling Rate (Hz):</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="samplingrate_input">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>2147483647</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="FrequencyGroup">
        <property name="title">
         <string>Frequency Info</string>
        </property>
        <layout class="QFormLayout" name="DataFrequencyGroupLayout">
         <property name="verticalSpacing">
          <number>3</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>3</number>
         </property>
         <item row="1" column="0">
          <widget class="QLabel" name="startfreq_label">
           <property name="text">
            <string>Start Frequency (Hz)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="startfreq_input">
           <property name="minimum">
            <number>-2147483647</number>
           </property>
           <property name="maximum">
            <number>2147483647</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="centerfreq_label">
           <property name="text">
            <string>Center Frequency (Hz)</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="centerfreq_input">
           <property name="minimum">
            <number>-2147483647</number>
           </property>
           <property name="maximum">
            <number>2147483647</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="stopfreq_label">
           <property name="text">
            <string>Stop Frequency (Hz)</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="stopfreq_input">
           <property name="minimum">
            <number>-2147483647</number>
           </property>
           <property name="maximum">
            <number>2147483647</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="TimeGroup">
        <property name="title">
         <string>Time Info</string>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>3</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="starttime_label">
           <property name="text">
            <string>Start Time</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDateTimeEdit" name="starttime_input">
           <property name="displayFormat">
            <string>dd/M/yyyy hh:mm:ss AP</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="stoptime_label">
           <property name="text">
            <string>Stop Time</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDateTimeEdit" name="stoptime_input">
           <property name="displayFormat">
            <string>dd/M/yyyy hh:mm:ss AP</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="duration_label">
           <property name="text">
            <string>Duration (s)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="duration">
           <property name="text">
            <string>dur</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="STFTParamsGroup">
        <property name="title">
         <string>STFT Parameters</string>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <property name="verticalSpacing">
          <number>3</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>3</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="nfft_label">
           <property name="text">
            <string>NFFT</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QWidget" name="fastlencontainer" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout">
            <property name="spacing">
             <number>7</number>
            </property>
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QSpinBox" name="nfft_input">
              <property name="maximum">
               <number>99999999</number>
              </property>
              <property name="value">
               <number>2048</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="prevfastlen_btn">
              <property name="toolTip">
               <string>Previous length that is a multiple of 2, 3, 5, 7. Good for fast FFT</string>
              </property>
              <property name="text">
               <string>Prev</string>
              </property>
              <property name="iconSize">
               <size>
                <width>20</width>
                <height>15</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="nextfastlen_btn">
              <property name="toolTip">
               <string>Next length that is a multiple of 2, 3, 5, 7. Good for fast FFT</string>
              </property>
              <property name="text">
               <string>Next</string>
              </property>
              <property name="iconSize">
               <size>
                <width>20</width>
                <height>15</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="freqresolution_label">
           <property name="text">
            <string>Freq Res (Hz)</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="freqresolution_input">
           <property name="maximum">
            <double>99999.990000000005239</double>
           </property>
           <property name="value">
            <double>100.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="stride_label">
           <property name="text">
            <string>Stride</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QWidget" name="stridecontainer" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <property name="leftMargin">
             <number>5</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>5</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QSpinBox" name="stride_input">
              <property name="maximum">
               <number>999999</number>
              </property>
              <property name="value">
               <number>1024</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="stride_fit_btn">
              <property name="text">
               <string>Fit Width</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="stride_halfnfft">
              <property name="text">
               <string>NFFT/2</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="stftsize_label">
           <property name="text">
            <string>Results</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QLabel" name="stftsize">
           <property name="text">
            <string>Height:   Width: </string>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="dynamicrange_label">
           <property name="text">
            <string>Dyn. Range (dB)</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QWidget" name="dynamicrange_container" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QSlider" name="dynamicrange_input">
              <property name="minimum">
               <number>10</number>
              </property>
              <property name="maximum">
               <number>70</number>
              </property>
              <property name="value">
               <number>70</number>
              </property>
              <property name="orientation">
               <enum>Qt::Orientation::Horizontal</enum>
              </property>
              <property name="invertedAppearance">
               <bool>false</bool>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TickPosition::NoTicks</enum>
              </property>
              <property name="tickInterval">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="dynamicrange_value">
              <property name="text">
               <string>70dB</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="FileInfoGroupBox">
        <property name="title">
         <string>File Info</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>3</number>
         </property>
         <item row="1" column="0">
          <widget class="QLabel" name="countlabel">
           <property name="text">
            <string>Total Files:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="sizelabel">
           <property name="text">
            <string>Total Size: </string>
           </property>
          </widget>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QTableWidget" name="urltable">
           <property name="editTriggers">
            <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
           </property>
           <property name="showGrid">
            <bool>false</bool>
           </property>
           <property name="cornerButtonEnabled">
            <bool>false</bool>
           </property>
           <property name="columnCount">
            <number>2</number>
           </property>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <column>
            <property name="text">
             <string>filesize</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>filepath</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="confirm_button">
        <property name="text">
         <string>Confirm</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="SpectrumViewer" name="spectrumviewer" native="true"/>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SpectrumViewer</class>
   <extends>QWidget</extends>
   <header>include/spectrumviewer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
